### PR TITLE
simplify stuff

### PR DESCRIPTION
**Pull request by michalszynkiewicz**
added abstraction for config, simplified creating new service discoveries and load balancers

### CODE REVIEW

File: api/src/main/java/io/smallrye/dux/Dux.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        return providers.stream().map(ServiceLoader.Provider::get)
+                .collect(Collectors.toMap(ElementWithType::type, Function.identity()));
+    }
+
+    private static final Dux dux = new Dux();
+
+    public static Dux getInstance() {
+        return dux;
+    }
+}
```
Comment (aureamunoz) [inline api/src/main/java/io/smallrye/dux/Dux.java:104]: I think we should add some unit tests in the api module, wdyt? May be in a differente PR?

File: api/src/main/java/io/smallrye/dux/spi/ServiceDiscoveryProvider.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package io.smallrye.dux.spi;
```
Comment (aureamunoz) [inline api/src/main/java/io/smallrye/dux/spi/ServiceDiscoveryProvider.java:1]: What means `spi`?

**Review by aureamunoz — APPROVED** [review]
A minor remark and a question about unit tests in the `api` module but, in general, looks good to me

File: api/src/main/java/io/smallrye/dux/spi/ServiceDiscoveryProvider.java (hunk 2 above)
Comment (michalszynkiewicz) [reply api/src/main/java/io/smallrye/dux/spi/ServiceDiscoveryProvider.java:1]: spi is for people who implement new service discovery and load balancer strategies. It stands for Service Provider Interface

**Pull request merged**